src/main.rs

```diff
@@ -62,6 +63,13 @@ fn interp(prg: &[u8], map: Vec<usize>) {
     let mut ptr = 0;
     let mut cells = Vec::with_capacity(30000);
     cells.resize(30000, 0u8);
+    let stdin = {
+        let mode = CString::new("r").unwrap();
+        let mode_raw = mode.into_raw();
+        let stdin = unsafe { fdopen(libc::STDIN_FILENO, mode_raw) };
```

Comment:
How about using `fdopen` to convert `STDIN_FILENO` to a file, instead of opening stderr a second time?

> The fdopen() function associates a stream with the existing file descriptor fildes.

Reply:
Huh?

Reply:
I'm an idiot. Disregard this thread.